app/src/processing/app/Editor.java

```diff
@@ -1265,43 +1260,58 @@ public void actionPerformed(ActionEvent e) {
     return menu;
   }
 
+  /**
+   * Wrapper around RSyntaxTextArea's RecordableTextActions. Normally,
+   * these actions use the event source, or the most recently focused
+   * component to find the text area to work on, but this does not
+   * always work. This wrapper makes them always work on the currently
+   * selected tab instead.
+   */
+  class RstaActionWrapper extends ActionWrapper {
+    RstaActionWrapper(RecordableTextAction action) {
+      super(action);
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+      ((RecordableTextAction) getWrappedAction()).actionPerformedImpl(e, getCurrentTab().getTextArea());
+    }
+  }
 
   private JMenu buildEditMenu() {
     JMenu menu = new JMenu(tr("Edit"));
     menu.setName("menuEdit");
     menu.setMnemonic(KeyEvent.VK_E);
 
-    undoItem = newJMenuItem(tr("Undo"), 'Z');
+    // Create a dummy RTextArea, to force it to create the static list
+    // of actions accessible through RTextArea.getAction().
+    new RTextArea();
+
+    RecordableTextAction undoAction = RTextArea.getAction(RTextArea.UNDO_ACTION);
+    JMenuItem undoItem = new JMenuItem(new RstaActionWrapper(undoAction));
```

Comment:
thanks for the clarification, inedeed the fact that RSTA handle automatically the enable/disable is a point.

Another thing that comes to mind is how the localization is handled? is it something already in the RSTA resources? about `getUndoPresentationName()` it seems that the menu is always presented as "Undo" or "Redo", IMHO it's something that can be easily dropped in case.